MapboxNavigation/CarPlayManager.swift

```diff
@@ -55,8 +55,19 @@ public class CarPlayManager: NSObject {
         }
     }
 
+    var trackingStateObservation: NSKeyValueObservation?
+    
     public fileprivate(set) var mainMapTemplate: CPMapTemplate?
-    public fileprivate(set) weak var currentNavigator: CarPlayNavigationViewController?
+    public fileprivate(set) weak var currentNavigator: CarPlayNavigationViewController? {
+        didSet {
+            if let controller = currentNavigator {
+                trackingStateObservation = controller.observe(\.tracksUserCourse) { [weak self] (controller, change) in
```

Comment:
Do we need to invalidate the NSKeyValueObservation at some point, at least on older iOS versions?

Reply:
Good catch. It would leak on iOS <= 10